backend/test/database.cpp

```diff
@@ -0,0 +1,102 @@
+#define CATCH_CONFIG_MAIN
+#include <catch2/catch_test_macros.hpp>
+#include <fstream>
+#include "database.hpp"
+
+std::string path = "test.db";
+std::string failpath = "random.db";
+std::ofstream file(path);
+
+TEST_CASE("Test Database class")
+{
+    REQUIRE(file.is_open());
+
+    SECTION("Create, insert, and select queries")
+    {
+        // open database
+        database::Database db = database::Database(path);
+        REQUIRE_NOTHROW(db.execute("CREATE TABLE names (name TEXT);"));
+
+        // insert names into the database
+        std::vector<std::string> names = {"Cedric", "Stella", "Jessica"};
+        for (const auto &name : names)
+        {
+            REQUIRE_NOTHROW(db.execute("INSERT INTO names (name) VALUES ('" + name + "');"));
+        }
+
+        // check that the names are in the database
+        std::function<std::string(sqlite3_stmt *)> callback = [](sqlite3_stmt *stmt)
+        {
+            return std::string(reinterpret_cast<const char *>(sqlite3_column_text(stmt, 0)));
+        };
+        std::vector<std::string> result = db.execute("SELECT * FROM names;", callback);
+        CHECK(result == names);
+
+        db.close();
+    }
+
+    SECTION("Reopening a database has persistent storage")
```

Comment:
**nitpick**: Keep the description tense consistent: "Access persistent storage in database"